pkg/objstore/client/factory.go

```diff
@@ -3,39 +3,58 @@ package client
 import (
 	"context"
 	"fmt"
-	"runtime"
 
-	"cloud.google.com/go/storage"
 	"github.com/go-kit/kit/log"
 	"github.com/improbable-eng/thanos/pkg/objstore"
 	"github.com/improbable-eng/thanos/pkg/objstore/gcs"
 	"github.com/improbable-eng/thanos/pkg/objstore/s3"
 	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/common/version"
-	"google.golang.org/api/option"
+	yaml "gopkg.in/yaml.v2"
 )
 
-var ErrNotFound = errors.New("no valid GCS or S3 configuration supplied")
+type objProvider string
+
+const (
+	GCS objProvider = "GCS"
+	S3  objProvider = "S3"
+)
+
+type BucketConfig struct {
+	Type    objProvider `yaml:"type"`
+	Content interface{} `yaml:"content"`
```

Comment:
Maybe just `config` instead of `content`?

Reply:
+1